Server crashes when dropping decorated pot

```
panic: interface conversion: interface {} is []string, not []interface {} [recovered]
        panic: interface conversion: interface {} is []string, not []interface {}

goroutine 62 [running]:
github.com/df-mc/dragonfly/server/session.(*Session).handlePackets.func1()
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/session/session.go:303 +0x69
panic({0x99fb40?, 0xc003604f90?})
        /home/anthony/go/go1.21.2/src/runtime/panic.go:914 +0x21f
github.com/df-mc/dragonfly/server/block.DecoratedPot.DecodeNBT({0x0, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}}}, 0x4?)
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/block/decorated_pot.go:86 +0x285
github.com/df-mc/dragonfly/server/internal/nbtconv.readItemStack(0x0?, 0x0?)
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/internal/nbtconv/read.go:206 +0x209
github.com/df-mc/dragonfly/server/internal/nbtconv.Item(0xc003604f00, 0x0)
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/internal/nbtconv/read.go:170 +0xae
github.com/df-mc/dragonfly/server/entity.ItemBehaviourConfig.New({0x0?, 0x0?, 0x0?, 0x0?}, {0x1bb6, {0xe45ae0, 0xc0012970e0}, 0x1, {0x0, 0x0}, ...})
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/entity/item_behaviour.go:32 +0x290
github.com/df-mc/dragonfly/server/entity.NewItemPickupDelay({0x1bb6, {0xe45ae0, 0xc0012970e0}, 0x1, {0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...}, ...)
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/entity/item.go:26 +0x87
github.com/df-mc/dragonfly/server/player.(*Player).Drop(0xc000b3e000, {0x1bb6, {0xe45ae0, 0xc0012970e0}, 0x1, {0x0, 0x0}, {0x0, 0x0, 0x0}, ...})
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/player/player.go:2160 +0x1f9
github.com/df-mc/dragonfly/server/session.(*InventoryTransactionHandler).handleNormalTransaction(0xc000039c40?, 0xc000240a00?, 0xc000240a00)
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/session/handler_inventory_transaction.go:106 +0x89b
github.com/df-mc/dragonfly/server/session.(*InventoryTransactionHandler).Handle(0x98a240?, {0xe4a8d8?, 0xc003405b30}, 0xc000240a00)
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/session/handler_inventory_transaction.go:24 +0x97
github.com/df-mc/dragonfly/server/session.(*Session).handlePacket(0xc000240a00, {0xe4a8d8, 0xc003405b30})
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/session/session.go:432 +0x8d
github.com/df-mc/dragonfly/server/session.(*Session).handlePackets(0xc000240a00)
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/session/session.go:312 +0xdc
created by github.com/df-mc/dragonfly/server/session.(*Session).Start in goroutine 1
        /home/anthony/go/pkg/mod/github.com/df-mc/dragonfly@v0.9.9-0.20230930211047-d519f75b5eb0/server/session/session.go:217 +0xa5

Process finished with the exit code 2
```

--- a/server/block/decorated_pot.go
+++ b/server/block/decorated_pot.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"github.com/df-mc/dragonfly/server/block/cube"
 	"github.com/df-mc/dragonfly/server/block/model"
+	"github.com/df-mc/dragonfly/server/internal/nbtconv"
 	"github.com/df-mc/dragonfly/server/item"
 	"github.com/df-mc/dragonfly/server/world"
 	"github.com/go-gl/mathgl/mgl64"
@@ -82,8 +83,8 @@ func (p DecoratedPot) EncodeNBT() map[string]any {
 // DecodeNBT ...
 func (p DecoratedPot) DecodeNBT(data map[string]any) any {
 	p.Decorations = [4]PotDecoration{}
-	if sherds, ok := data["sherds"]; ok {
-		for i, name := range sherds.([]any) {
+	if sherds := nbtconv.Slice(data, "sherds"); sherds != nil {
+		for i, name := range sherds {
 			it, ok := world.ItemByName(name.(string), 0)
 			if !ok {
 				panic(fmt.Errorf("unknown item %s", name))

